### DISCUSSION TITLE
s3 backend?

### DISCUSSION THREAD

**Discussion by gytis-ivaskevicius** [transmission/transmission#5084] [Ideas]
It would be cool if transmission would support S3 for storage natively

And yes, I know that one can just fuse mount s3

**Comment by nevack** [comment]
I think this is out of scope of this project, @ckerr will tell if i'm right or not.
Also, is it possible to read/write data with S3 at random offsets?

**Reply by gytis-ivaskevicius** [reply]
It is possible to read at the offset but not write. I would imagine implementation having a working dir for downloads and once it is downloaded auto-submitted to s3

Now that I think about it - one may implement it using via post download hook :thinking:

**Reply by nevack** [reply]
Yes, looks like you can do it with post-download script hook.

**Comment by ckerr** [comment]
@nevack is right.

There are a lot of suggestions like this one that are kind of cool (and TBH I would probably use this S3 backend feature) but the bottom line is that maybe 0.1% of all Transmission users would benefit from this.

Since few users would use it and since every feature adds to code complexity and maintenance cost, it doesn't make sense to add edge cases like this to the codebase. :man_shrugging:

**Comment by groceryheist** [comment]
Have you tried mounting an S3 bucket with Rclone as your transmission data directory?
